oteapi/models/genericconfig.py

```diff
@@ -0,0 +1,60 @@
+"""Generic data model for configuration attributes."""
+from typing import Any, Optional
+
+from pydantic import BaseModel, Field
+
+
+class AttrDict(BaseModel):
+    """An object whose attributes can also be accessed through
+    subscription, like with a dictionary."""
+
+    class Config:
+        """Class for configuration of pydantic models."""
+
+        extra = "allow"
+
+    def __contains__(self, name: Any) -> bool:
+        """Enable using the 'in' operator on this object."""
+        return self.__dict__.__contains__(name)
+
+    def __delitem__(self, key: str) -> None:
+        """Enable deletion access through subscription."""
+        if key in dir(self):
+            self.__delattr__(key)
+        else:
+            raise KeyError(key)
+
+    def __getitem__(self, key: str) -> Any:
+        """Enable read access through subscription."""
+        if key in dir(self):
+            return getattr(self, key)
+        raise KeyError(key)
+
+    def __setitem__(self, key: str, value: Any) -> None:
+        """Enable write access through subscription."""
+        if key in self.__dict__:
+            target_type = type(self.__dict__[key])
+            if not isinstance(value, target_type):
+                raise TypeError(
+                    "Mapped value must be subclass of " + target_type.__name__
+                )
+        self.__dict__[key] = value
+
+
+class GenericConfig(AttrDict):
```

Comment:
```suggestion
class GenericConfig(BaseModel):
```
Not sure I'm for making the `GenericConfig` a wannabe dictionary. However, I'm not completely against it either. But for sure it shouldn't allow random extras... I think. That's for the `configuration` only.
Therefore, I think it's easier to use `BaseModel` here; we want the strict attribute validation for the config classes.

Reply:
If ´DataCacheConfig´ is the only `Config` class that will ever be used as a `configuration` field, then the other `Config` classes don't need this subscripting flexibility, so then `BaseModel` suffices for them. I've reverted `GenericConfig's` parent to `BaseModel`.